ros2_batch_job/util.py

```diff
@@ -167,12 +167,28 @@ def warn(*args, **kwargs):
 
 class MyProtocol(AsyncSubprocessProtocol):
     def __init__(self, cmd, exit_on_error, *args, **kwargs):
-        self.cmd = cmd
+        self.cmd = cmd  
         self.exit_on_error = exit_on_error
+        self.progress_bar = False
+        self.progress = []
         AsyncSubprocessProtocol.__init__(self, *args, **kwargs)
 
     def on_stdout_received(self, data):
-        sys.stdout.write(data.decode('utf-8', 'replace').replace(os.linesep, '\n'))
+        if b'[?25l' in data:
+            self.progress_bar = True
+        if b'[?25h' in data:
+            if self.progress:
```

Comment:
This `if` statement isn't required; the loop below will execute 0 times if there is no data in `self.progress`.